### PR TITLE
[RF] Fix dependencies for RooFit tests for out-of-root builds.

### PR DESCRIPTION
**Pull request by hageboeck**

### CODE REVIEW

File: root/roofitstats/vectorisedPDFs/CMakeLists.txt [hunk 1]
```diff
@@ -6,27 +6,46 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL Intel AND DEFINED ENV{VTUNE_AMPLIFIER_2019_DIR
 endif()
 
 add_library(VectorisedPDFTests STATIC VectorisedPDFTests.cxx)
-link_libraries(VectorisedPDFTests Gpad RooFitCore RooFit)
-add_dependencies(VectorisedPDFTests RooFitCore)
+target_link_libraries(VectorisedPDFTests Gpad RooFitCore RooFit)
```
Comment (amadio) [inline root/roofitstats/vectorisedPDFs/CMakeLists.txt]: This is better, but will only work if you use `ROOT::...` for the library names.

Comment (hageboeck) [reply root/roofitstats/vectorisedPDFs/CMakeLists.txt]: Is listing the libraries preferred?

**Comment by hageboeck** [comment]
@linev Try again. I used the `ROOT::targetname` that @amadio mentioned.

**Comment by linev** [comment]
> Try again. I used the ROOT::targetname that @amadio mentioned.

Sorry, still not working for me. 
Did you try yourself? 
Just call `thisroot.sh` and then `cmake path/to/roottest`.

File: root/roofitstats/vectorisedPDFs/CMakeLists.txt [hunk 2]
```diff
@@ -6,27 +6,46 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL Intel AND DEFINED ENV{VTUNE_AMPLIFIER_2019_DIR
 endif()
 
 add_library(VectorisedPDFTests STATIC VectorisedPDFTests.cxx)
-link_libraries(VectorisedPDFTests Gpad RooFitCore RooFit)
-add_dependencies(VectorisedPDFTests RooFitCore)
+target_link_libraries(VectorisedPDFTests ROOT::Gpad ROOT::RooFitCore ROOT::RooFit)
```
Comment (amadio) [inline root/roofitstats/vectorisedPDFs/CMakeLists.txt]:
```suggestion
target_link_libraries(VectorisedPDFTests PUBLIC ROOT::Gpad ROOT::RooFitCore ROOT::RooFit)
```

File: root/roofitstats/vectorisedPDFs/CMakeLists.txt (hunk 1 above)
Comment (amadio) [reply root/roofitstats/vectorisedPDFs/CMakeLists.txt]: Either is fine, using the targets is probably easier.

**Comment by linev** [comment]
Still does not work :(

**Comment by hageboeck** [comment]
Ok, need time to test after meeting

**Comment by hageboeck** [comment]
The problem is a forgotten
```
 794   if(CMAKE_PROJECT_NAME STREQUAL ROOT)
 795     add_dependencies(${library} move_headers)
 796   endif()
```
in `root/cmake/modules/RootMacros.cmake `
Since nobody used gtests in roottest before, it was never found. I can open a PR to fix that since Guilherme will be out of the country in the coming weeks.

**Comment by hageboeck** [comment]
https://github.com/root-project/root/pull/4367

**Comment by linev** [comment]
With https://github.com/root-project/root/pull/4367 it works for me! 
But this PR is still necessary.

**Pull request merged**